packages/core/src/publicCredential/PublicCredential.ts

```diff
@@ -0,0 +1,206 @@
+/**
+ * Copyright (c) 2018-2022, BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+import type { HexString } from '@polkadot/util/types'
+import type { AccountId } from '@polkadot/types/interfaces'
+import type { PublicCredentialsCredentialsCredential } from '@kiltprotocol/augment-api'
+import type {
+  DidUri,
+  IAssetClaim,
+  ICType,
+  IDelegationNode,
+  IPublicCredential,
+  IPublicCredentialInput,
+  PartialAssetClaim,
+} from '@kiltprotocol/types'
+
+import { blake2AsHex } from '@polkadot/util-crypto'
+import { ConfigService } from '@kiltprotocol/config'
+import * as AssetDid from '@kiltprotocol/asset-did'
+import * as Did from '@kiltprotocol/did'
+import { DataUtils, SDKErrors } from '@kiltprotocol/utils'
+
+import { verifyClaimAgainstSchema } from '../ctype/CType.js'
+import { toChain as publicCredentialToChain } from './PublicCredential.chain.js'
+
+/**
+ * Calculates the ID of a [[IPublicCredentialInput]], to be used to retrieve the full content from the blockchain.
+ *
+ * The ID is formed by first concatenating the SCALE-encoded [[IPublicCredentialInput]] with the SCALE-encoded [[DidUri]] and then Blake2b hashing the result.
+ *
+ * @param credential The input credential object.
+ * @param attester The DID of the credential attester.
+ * @returns The credential ID.
+ */
+export function computeId(
+  credential: IPublicCredentialInput,
+  attester: DidUri
+): HexString {
+  const api = ConfigService.get('api')
+
+  const scaleEncodedCredential = api
+    .createType<PublicCredentialsCredentialsCredential>(
+      'PublicCredentialsCredentialsCredential',
+      publicCredentialToChain(credential)
+    )
+    .toU8a()
+  const scaleEncodedAttester = api
+    .createType<AccountId>('AccountId', Did.toChain(attester))
+    .toU8a()
+
+  return blake2AsHex(
+    Uint8Array.from([...scaleEncodedCredential, ...scaleEncodedAttester])
+  )
+}
+
+function verifyClaimsStructure(input: IAssetClaim | PartialAssetClaim): void {
```

Comment:
Should it remain "verify claimS structure"?

Reply:
Yeah makes sense: https://github.com/KILTprotocol/sdk-js/pull/677/commits/023ad1c369410f4be90a93e0c4363bf2b8759885#diff-0ed53062bda598489034fe7ca13c690382a38738d8e6ea23ebfc8b3a6a3b5880R60